### PR TITLE
Add checkstyle, PMD, findbugs

### PR DESCRIPTION
**Pull request by amlcurran**
Add our standard set of code analysis to the project. We've also replaced the previous checkstyle implementation with the official plug in, rather than our own custom implementation

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/24/
Test PASSed.

**Comment by xrigau** [comment]
don't forget we'll have to add the reports file paths to jenkins!

**Comment by amlcurran** [comment]
:+1: good point

File: code-quality/android-code-quality.gradle [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+def sourceSetMain = android.sourceSets.main.java.srcDirs
+def rulesDir = new File("$project.rootDir/code-quality", 'rules')
+def ignoreFail = true // sad panda
+
+def includeSrcPatternApplication = 'com/novoda/downloadmanager/**/*.java'
+def excludeSrcPatternRClass = 'com/novoda/downloadmanager/**/R.java'
+def excludeSrcPatternGeneratedFiles = '**/gen/**'
+
+apply plugin: 'checkstyle'
+task checkstyleMain(type: Checkstyle) {
```
Comment (blundell) [inline code-quality/android-code-quality.gradle]: can we still have these as 3 separate files so we can easily distinguish them and change them separately?

Comment (amlcurran) [inline code-quality/android-code-quality.gradle]: we can indeed

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/26/
Test PASSed.

File: code-quality/android-code-quality.gradle (hunk 1 above)
Comment (amlcurran) [inline code-quality/android-code-quality.gradle]: https://github.com/novoda/download-manager/commit/b33971516c937050b090c8a5fdfe5f446969b019

**Comment by blundell** [comment]
LGTM, so we can use this because we don't have a /core module is that correct?or is it because of some other change in gradle versions

**Comment by amlcurran** [comment]
not sure, this configuration was taken from another project and this from their mobile module, but they have a similar config in core as well

**Comment by xrigau** [comment]
@blundell this is how cool kids do it now, use the standard gradle plugins for both java and android modules (but android need some more setup) :space_invader:

**Comment by xrigau** [comment]
this is ready to :ship: #justsayin

**Comment by ouchadam** [comment]
let's do it

**Pull request merged**